### PR TITLE
Use bytecode for comparing whether to add/remove event listeners

### PR DESCRIPTION
**Pull request by berendkleinhaneveld**
Possible fix for #4 .

- [x] Use context in comparison
- [x] Add tests
- [x] Fix test with closures

### CODE REVIEW

**Comment by Korijn** [comment]
I can't help but wonder if this really is the right problem to solve. How do vue and react deal with this?

**Comment by Korijn** [comment]
I'm worried that you might end up keeping references to old, outdated variables from functional components with this approach. Can you add a test to check this?

File: pygui/__init__.py [hunk 1]
```diff
@@ -527,3 +533,57 @@ def create_ops(current, future):
...
+        "co_nlocals",
+        "co_posonlyargcount",
+        "co_stacksize",
+    ]:
+        attr_a = getattr(a, attr, None)
+        attr_b = getattr(b, attr, None)
+        if attr_a != attr_b:
+            return False
+
+    return True
```
Comment (Korijn) [inline pygui/__init__.py:589]: just checking: is re-evaluating this on every render really more efficient than re-attaching event handlers on every render?

Comment (berendkleinhaneveld) [reply pygui/__init__.py:589]: Yes, apparently it still outperformed removing the handler and attaching the new one.

Comment (berendkleinhaneveld) [reply pygui/__init__.py:589]: I tested this on the pygfx example, but the actual answer should be: it depends :) It depends on the specific renderer.

**Comment by Korijn** [comment]
I'n impressed :) didn't realize you could just access the closure in Python, but it seems only natural now that I see it in action! Awesome!

**Comment by berendkleinhaneveld** [comment]
I found these two awesome resources:

* https://towardsdatascience.com/understanding-python-bytecode-e7edaae8734d
* https://towardsdatascience.com/closures-and-decorators-in-python-2551abbc6eb6

Great introduction into bytecode and closures

**Comment by ivoflipse** [comment]
I do wonder how stabile these attributes are between Python versions. Maybe the Python API for checking bytecode is, but I do know these get mucked around with a lot

**Comment by berendkleinhaneveld** [comment]
Yes, that's certainly a possible problem, but that's where CI can help. Seems that Python 3.8-3.10 pass the tests. As long as those attributes are available and comparable it should be fine. Given that most of these items are tuples that should not be a problem.

**Comment by Korijn** [comment]
Can we merge?

**Comment by berendkleinhaneveld** [comment]
I think so, yes!

**Pull request merged**